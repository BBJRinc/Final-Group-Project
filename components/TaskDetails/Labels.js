import React, { Component } from 'react';
import { Modal, TouchableHighlight } from 'react-native';
import { Container, Header, Left, Body, Right, Button, Title, Text, Input, Item, Content, Form, View } from 'native-base';
import IconI from 'react-native-vector-icons/Ionicons'
import IconE from 'react-native-vector-icons/Entypo'
import IconF from 'react-native-vector-icons/Feather'


export default class Labels extends Component {
    state = {
        modalVisible: false,
    };

    setModalVisible(visible) {
        this.setState({ modalVisible: visible });
    }
    componentWillReceiveProps() {
        this.setState({modalVisible:this.props.isVisable})
    }

    render() {
        console.log(this.state)
        const { LabelPadding, headerText, colorMargin, buttonStyle, labelStyle } = styles
        return (
            <View style={{ marginTop: 22 }}>
                <Modal
                    animationType="slide"
                    transparent={false}
                    visible={this.state.modalVisible}
                    >
                    <Header style={{ backgroundColor: this.props.color }}>
                        <Left>
                            <Button transparent >
                                <TouchableHighlight 
                                    onPress={() => {
                                        this.setModalVisible(false);
                                    }}>
                                    <IconI name='ios-close' size={30} color={'#fff'} />
                                </TouchableHighlight>
                            </Button>
                        </Left>
                        <Text style={headerText}>Label</Text>
                        <Right>
                            <Button transparent>
                                <IconE name='dots-three-horizontal' size={20} color={'#fff'} />
                            </Button>
                        </Right>
                    </Header>
                    <Content>
                        <Content>
                            <Button block style={[buttonStyle, { backgroundColor: '#EB5A46' }]} onPress={(e) => this.props.labelColor({ color: '#EB5A46', modalVisible:'false'})}>{this.props.color === '#EB5A46' ? <IconF name='check' size={25} style={{ color: '#fff' }} /> : null}</Button>
                            <Button block style={[buttonStyle, { backgroundColor: '#FFAB4A' }]} onPress={(e) => this.props.labelColor({ color: '#FFAB4A', modalVisible:'false' })}>{this.props.color === '#FFAB4A' ? <IconF name='check' size={25} style={{ color: '#fff' }} /> : null}</Button>
                            <Button block style={[buttonStyle, { backgroundColor: '#51E898' }]} onPress={(e) => this.props.labelColor({ color: '#51E898', modalVisible:'false' })}>{this.props.color === '#51E898' ? <IconF name='check' size={25} style={{ color: '#fff' }} /> : null}</Button>
                            <Button block style={[buttonStyle, { backgroundColor: '#F5DD29' }]} onPress={(e) => this.props.labelColor({ color: '#F5DD29', modalVisible:'false' })}>{this.props.color === '#F5DD29' ? <IconF name='check' size={25} style={{ color: '#fff' }} /> : null}</Button>
                            <Button block style={[buttonStyle, { backgroundColor: '#00C2E0' }]} onPress={(e) => this.props.labelColor({ color: '#00C2E0', modalVisible:'false' })}>{this.props.color === '#00C2E0' ? <IconF name='check' size={25} style={{ color: '#fff' }} /> : null}</Button>
                            <Button block style={[buttonStyle, { backgroundColor: '#838C91' }]} onPress={(e) => this.props.labelColor({ color: '#838C91', modalVisible:'false' })}>{this.props.color === '#838C91' ? <IconF name='check' size={25} style={{ color: '#fff' }} /> : null}</Button>
                        </Content>
                    </Content>
                </Modal>
                <TouchableHighlight style={{alignItems:'center'}}
                    onPress={() => {
                        this.setModalVisible(true);
                    }}>
                    <Text style={LabelPadding}>{this.props.color==='' ? 'Label...' : <View style={[labelStyle,{backgroundColor:this.props.color}]}><Text style={{color:'#fff', paddingLeft:10, justifyContent:'center' }}>Tap to change task color</Text></View>}</Text>
                </TouchableHighlight>
            </View>
        );
    }
}
const styles = ({
    LabelPadding: {
        height: 40,
        marginLeft: 10,
        color: '#C7C7CD',
        alignItems: 'center',
        justifyContent:'center',
        alignContent:'center',
    },
    headerText: {
        color: '#fff',
        paddingTop: 13
    },
    colorMargin: {
        marginTop: 5
    },
    buttonStyle: {
        height: 50,
        width: 300,
        margin: 5,
        justifyContent: 'flex-end',
        paddingRight: 15
    }, 
    labelStyle:{
        height:25, 
        width: 225, 
        borderRadius:3, 
        marginLeft:10, 
        alignItems:'center'
    }
})