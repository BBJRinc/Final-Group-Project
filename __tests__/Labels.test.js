//Brandons Test

import React from 'react';
import Labels from '../components/TaskDetails/Labels'

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Labels />).toJSON();
    expect(tree).toMatchSnapshot();
});


