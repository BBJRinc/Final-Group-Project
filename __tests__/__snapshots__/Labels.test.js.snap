// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "marginTop": 22,
      },
    ]
  }
  virtual={undefined}
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderBottomColor": "#a7a6ab",
              "borderBottomWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 15,
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": undefined,
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 6,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 30,
                    },
                    null,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#fff",
                "paddingTop": 13,
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          Label
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 50,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginRight": -8,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Entypo",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={undefined}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      viewIsInsideTabBar={false}
      virtual={undefined}
    >
      <View>
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={undefined}
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutoAutomaticScroll={true}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          getScrollResponder={[Function]}
          handleOnScroll={[Function]}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          onScroll={[Function]}
          resetKeyboardSpace={[Function]}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          scrollForExtraHeightOnAndroid={[Function]}
          scrollToEnd={[Function]}
          scrollToFocusedInput={[Function]}
          scrollToPosition={[Function]}
          showsVerticalScrollIndicator={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
          viewIsInsideTabBar={false}
          virtual={undefined}
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#EB5A46",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "flex-end",
                  "margin": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "width": 300,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#FFAB4A",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "flex-end",
                  "margin": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "width": 300,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#51E898",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "flex-end",
                  "margin": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "width": 300,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#F5DD29",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "flex-end",
                  "margin": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "width": 300,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#00C2E0",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "flex-end",
                  "margin": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "width": 300,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#838C91",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "flex-end",
                  "margin": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "width": 300,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            />
          </View>
        </RCTScrollView>
      </View>
    </RCTScrollView>
  </Modal>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "height": 25,
            "marginLeft": 10,
            "width": 225,
          },
          Object {
            "backgroundColor": undefined,
            "marginBottom": 20,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#fff",
              "justifyContent": "center",
              "paddingLeft": 10,
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        Tap to change task color
      </Text>
    </View>
  </View>
</View>
`;
